reactor-core/src/main/java/reactor/core/publisher/ImmutableSignal.java

```diff
@@ -82,49 +82,38 @@ public Context getContext() {
 	}
 
 	@Override
-	public boolean equals(@Nullable Object o) {
+	public boolean equals(Object o) {
```

Comment:
Can you elaborate on these apparent semantics change?

Reply:
my first iteration was adding the SUBSCRIBE and CANCEL cases to the `Signal` class, but that is likely to cause issues with existing open-ended `if-elseif-else` styles of dealing with the `Signal`. I've since reverted.